search-app/src/cmr/search/services/acls/collections_cache.clj

```diff
@@ -116,67 +116,34 @@
- ([context coll-concept-id]
-  (let [coll-by-concept-id-cache (hash-cache/context->cache context coll-for-gran-acl-caches/coll-by-concept-id-cache-key)
-        collection (hash-cache/get-value
-                    coll-by-concept-id-cache
-                    coll-for-gran-acl-caches/coll-by-concept-id-cache-key
-                    coll-concept-id)]
-   (when (or (nil? collection) (empty? collection))
-    (info (coll-msg/collection-not-found coll-concept-id))
-    (coll-for-gran-acl-caches/refresh-entire-cache context)) ;;TODO replace
-   (time-strs->clj-times collection)))
- ([context provider-id entry-title]
-  (let [coll-by-provider-id-and-entry-id-cache (hash-cache/context->cache context coll-for-gran-acl-caches/coll-by-provider-id-and-entry-title-cache-key)
-        collection (hash-cache/get-value
-                    coll-by-provider-id-and-entry-id-cache ;; cache
-                    coll-for-gran-acl-caches/coll-by-provider-id-and-entry-title-cache-key ;; key
-                    (str provider-id entry-title))]
-   (when (or (nil? collection) (empty? collection))
-    (info (coll-msg/collection-not-found (str provider-id entry-title)))
-    (coll-for-gran-acl-caches/refresh-entire-cache context)) ;;TODO replace
-   (time-strs->clj-times collection))))
-
-;; TODO remove after transition
-(defn- get-collections-map
-  "Gets the cached value. By default an error is thrown if there is no value in
-   the cache. Pass in anything other then :return-errors for the third parameter
-   to have null returned when no data is in the cache. :return-errors is for the
-   default behavior."
-  ([context key-type]
-   (get-collections-map context key-type :return-errors))
-  ([context key-type throw-errors]
-  (let [coll-cache (hash-cache/context->cache context cache-key)
-        collection-map (hash-cache/get-value
-                        coll-cache
-                        cache-key
-                        key-type)]
-    (if (and (= :return-errors throw-errors) (empty? collection-map))
-      (errors/internal-error! (coll-msg/collections-not-in-cache key-type))
-      (time-strs->clj-times collection-map)))))
-
-;; TODO remove after transition
-(defn get-collection
-  "Gets a single collection from the cache by concept id. Handles refreshing the cache if it is not found in it.
-  Also allows provider-id and entry-title to be used."
-  ([context concept-id]
-   (let [by-concept-id (get-collections-map context :by-concept-id :no-errors)]
-     (when-not (and by-concept-id (by-concept-id concept-id))
-       (info (coll-msg/collection-not-found concept-id))
-       (refresh-cache context))
-     (get (get-collections-map context :by-concept-id ) concept-id)))
+(defn get-collection-for-gran-acls
+  ([context coll-concept-id]
+   "Gets a single collection from the cache by concept id. If collection is not found, will add it to the cache (if it exists in elastic) and will return the found collection."
```

Comment:
Remove the word "not" from this sentence.  I think the negation is not true.

Reply:
I will clarify that if it's not found in cache but found in elastic

Reply:
fixed